py/objdict.c

```diff
@@ -290,7 +290,7 @@ STATIC mp_obj_t dict_get_helper(size_t n_args, const mp_obj_t *args, mp_map_look
         } else {
             value = args[2];
         }
-        if (lookup_kind == MP_MAP_LOOKUP_ADD_IF_NOT_FOUND) {
+        if (elem && lookup_kind == MP_MAP_LOOKUP_ADD_IF_NOT_FOUND) {
             elem->value = value;
```

Comment:
The semantics of `MP_MAP_LOOKUP_ADD_IF_NOT_FOUND` guarantee that the returned `elem` is non-NULL (because it's adding the elem if it couldn't find it).  So this would be better as an `assert(elem != NULL)` inside the if.